core/core.go

```diff
@@ -90,12 +90,6 @@ func RunBazeliskWithArgsFunc(argsFunc ArgsFunc, repos *Repositories) (int, error
 // RunBazeliskWithArgsFuncAndConfig runs the main Bazelisk logic for the given ArgsFunc and Bazel
 // repositories and config.
 func RunBazeliskWithArgsFuncAndConfig(argsFunc ArgsFunc, repos *Repositories, config config.Config) (int, error) {
-	return RunBazeliskWithArgsFuncAndConfigAndOut(argsFunc, repos, config, nil)
-}
-
-// RunBazeliskWithArgsFuncAndConfigAndOut runs the main Bazelisk logic for the given ArgsFunc and Bazel
-// repositories and config, writing its stdout to the passed writer.
-func RunBazeliskWithArgsFuncAndConfigAndOut(argsFunc ArgsFunc, repos *Repositories, config config.Config, out io.Writer) (int, error) {
```

Comment:
The removal of this function will break users (https://github.com/bazelbuild/bazelisk/pull/526), so I'd like to keep it.
Also, in the current version the out parameter of runBazel is basically useless since it will always be nil.

Reply:
Ah thanks, I didn't realize bazelisk supported use as a library so I just looked for uses in bazelisk itself.  Added it back.

Are there going to be callers who need to call one of these RunBazelisk functions with `out == nil` and have it return instead of `exec`ing?  The way I've written this right now doesn't support that, and I'm not sure how you'd rather me do that if we need that - duplicate all these public functions with ExecBazelisk copies?  Or add a flag to exec if possible or not?

Reply:
Ok I think if I were a user I'd want to have the behavior of these preserved to return instead of execing, so I've added `Exec` versions of the ones that make sense (without `out`) and left the `Run` ones to still run in a subprocess and return